stacks/spica/src/function/function.service.ts

```diff
@@ -34,19 +41,35 @@ export class FunctionService {
     return this.store.pipe(select(fromFunction.selectEntities)).pipe(map(entities => entities[id]));
   }
 
-  getLogs(filter: LogFilter): Observable<Log[]> {
-    const serializedFilter: {[P in keyof LogFilter]?: string | string[]} = {
-      functions: filter.functions
-    };
+  getLogs(
+    filter: LogFilter,
+    limit: number = 0,
+    skip: number = 0,
+    sort: object = {_id: -1}
+  ): Observable<Log[]> {
+    let queryParams: string[] = [];
+
+    if (filter.function.length > 0) {
+      filter.function.forEach(fn => queryParams.push(`functions=${fn}`));
+    }
 
     if (filter.begin instanceof Date) {
-      serializedFilter.begin = this.resetTimezoneOffset(filter.begin).toISOString();
+      queryParams.push(`begin=${this.resetTimezoneOffset(filter.begin).toISOString()}`);
     }
 
     if (filter.end instanceof Date) {
-      serializedFilter.end = this.resetTimezoneOffset(filter.end).toISOString();
+      queryParams.push(`end=${this.resetTimezoneOffset(filter.end).toISOString()}`);
     }
-    return this.http.get<any[]>(`api:/function/logs`, {params: serializedFilter});
+
+    queryParams.push(`limit=${limit}`);
+    queryParams.push(`skip=${skip}`);
+    queryParams.push(`sort=${JSON.stringify(sort)}`);
+
+    queryParams.push(`Authorization=${this.passport.token}`);
+
+    let mergedParams = "?" + queryParams.join("&");
```

Comment:
You could use the URL class to do this which is more appropriate to build up links like this.